src/resource-archive/index.ts

```diff
@@ -42,8 +49,21 @@ class Watcher {
   }
 
   async idle() {
-    // TODO -- wait for network idle
-    await new Promise((r) => setTimeout(r, 1000));
+    // Wait for network idle, but wait a maximum of 4s.
+    const ac: AbortController = new AbortController();
+
+    // eslint-disable-next-line @typescript-eslint/no-implied-eval
+    let timeoutId: any;
+    const timeoutPromise = new Promise((r) => {
+      timeoutId = setTimeout(r, 4000);
+    });
+
+    await Promise.race([
+      timeoutPromise,
+      Promise.all([...this.unfulfilledRequests.values(), ...this.unsettledResolvers.values()]),
```

Comment:
What happens if a request creates further requests in resolving?

I think probably we need to do something more complicated, ie check at the end of each request if "all requests are resolved".

Reply:
Ok, I can add this logic. I had another concern, too, that `Promise.all` might resolve _before_ any of the requests are actually added to the set. I suppose I could add another 500ms wait at the very beginning, but this seems like the incorrect solution.